xds/src/main/java/io/grpc/xds/internal/rbac/engine/GrpcAuthorizationEngine.java

```diff
@@ -316,9 +335,39 @@ private String getPath() {
 
     @Nullable
     private String getHeader(String headerName) {
-      if (headerName.endsWith(Metadata.BINARY_HEADER_SUFFIX)) {
+      if ("TE".equals(headerName)) {
```

Comment:
equalsIgnoreCase? Ditto below. Feel free to lower case `headerName` once at the beginning here.